ConsoleApp/Program.cs

```
@@ -0,0 +1,267 @@
+                    item = item.Next;
+                }
+
+                return array;
+            }
+            public void Clear()
+            {
+                _head = null;
+                Count = 0;
+            }
+            private T ShowAndRemove(int index)
+            {
+                T retValue;
+                if ((index == Count) || (index < 0))
+                {
+                    // throw new ArgumentOutOfRangeException(nameof(index));
+                    throw new Exception("Stack is Empty");
+                }
+
+                if (index == 0)
+                {
+                    var prev = GetByIndex(0);
+                    retValue = prev.Value;
+                    _head = _head.Next;
+                }
+                else
+                {
+                    var prev = GetByIndex(index - 1);
+                    retValue = prev.Next.Value;
+                    prev.Next = prev.Next.Next;
+                }
+
+                --Count;
+                return retValue;
+            }
+            private T Show(int index)
+            {
+                T retValue;
+                if ((index == Count) || (index < 0))
+                {
+                    throw new Exception("Stack is Empty");
+                    // throw new ArgumentOutOfRangeException(nameof(index));
+                }
+
+                if (index == 0)
+                {
+                    var prev = GetByIndex(0);
+                    retValue = prev.Value;
+                }
+                else
+                {
+                    var prev = GetByIndex(index - 1);
+                    retValue = prev.Next.Value;
+                }
+                return retValue;
+            }
+            public T Pop()
+            {
+                try
+                {
+                    return ShowAndRemove(Count - 1);
+                }
+                catch (Exception ex)
+                {
+                    return default(T);
```

if you have no elements it's better to throw this exception because your stack can have elements equal to the default